CMakeLists.txt

```diff
@@ -441,19 +456,44 @@ if (WIN32)
 endif (WIN32)
 
 if (UNIX)
-	set(PIONEER_DESKTOP_FILE ${CMAKE_BINARY_DIR}/metadata/net.pioneerspacesim.Pioneer.desktop)
-	configure_file(metadata/net.pioneerspacesim.Pioneer.desktop.cmakein ${PIONEER_DESKTOP_FILE} @ONLY)
-	install(FILES ${PIONEER_DESKTOP_FILE}
-		DESTINATION ${CMAKE_INSTALL_DATADIR}/applications
-	)
-	install(FILES metadata/net.pioneerspacesim.Pioneer.appdata.xml
-		DESTINATION ${CMAKE_INSTALL_DATADIR}/appdata
-	)
-
-	foreach(_i IN ITEMS 16 22 24 32 40 48 64 128 256)
-		install(FILES application-icon/pngs/pioneer-${_i}x${_i}.png
-			DESTINATION ${CMAKE_INSTALL_DATADIR}/icons/hicolor/${_i}x${_i}/apps
-			RENAME net.pioneerspacesim.Pioneer.png
+	if (APPIMAGE_BUILD)
+		set(PIONEER_DESKTOP_FILE ${CMAKE_BINARY_DIR}/appimage-metadata/net.pioneerspacesim.Pioneer.desktop)
+		set(PIONEER_SH_FILE ${CMAKE_BINARY_DIR}/init_pioneer.sh)
+		configure_file(appimage-metadata/net.pioneerspacesim.Pioneer.desktop.cmakein ${PIONEER_DESKTOP_FILE} @ONLY)
+		configure_file(init_pioneer.sh.cmakein ${PIONEER_SH_FILE} @ONLY)
+		install(FILES ${PIONEER_DESKTOP_FILE}
+			DESTINATION ${CMAKE_INSTALL_DATADIR}/applications
+		)
+		install(FILES ${PIONEER_SH_FILE}
+			DESTINATION ${CMAKE_INSTALL_BINDIR}
+			PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE GROUP_READ GROUP_WRITE GROUP_EXECUTE WORLD_READ WORLD_WRITE WORLD_EXECUTE
+		)
+		foreach(_i IN ITEMS 16 22 24 32 40 48 64 128 256)
+			install(FILES application-icon/pngs/pioneer-${_i}x${_i}.png
+				DESTINATION ${CMAKE_INSTALL_DATADIR}/icons/hicolor/${_i}x${_i}/apps
+				RENAME net.pioneerspacesim.Pioneer.png
+			)
+		endforeach()
+	else()
+		set(PIONEER_DESKTOP_FILE ${CMAKE_BINARY_DIR}/metadata/net.pioneerspacesim.Pioneer.desktop)
+		set(PIONEER_SH_FILE ${CMAKE_BINARY_DIR}/metadata/runpioneer.sh)
```

Comment:
What is `runpioneer.sh`? I don't see it anywhere. Why do we need this `runpioneer.sh` script now, in the non-AppImage case?

Reply:
I will fix it, seems like I missed removing this part

Reply:
fixed it